test/abort-controller.js

```diff
@@ -0,0 +1,210 @@
+'use strict'
+
+const { test } = require('tap')
+const { AbortController } = require('abort-controller')
+const { Client, errors } = require('..')
+const { createServer } = require('http')
+const { createReadStream } = require('fs')
+
+test('Abort while sending request - event emitter (no body)', { skip: 'never ending' }, (t) => {
+  t.plan(1)
+
+  const server = createServer((req, res) => {
+    t.fail('The requets should be aborted')
+  })
+  t.teardown(server.close.bind(server))
+
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`)
+    const abortController = new AbortController()
+    t.teardown(client.close.bind(client))
+
+    client.request({ path: '/', method: 'GET', signal: abortController.signal }, (err, response) => {
+      t.ok(err instanceof errors.RequestAbortedError)
+    })
+
+    abortController.abort()
```

Comment:
since you don't have a body, the request gets dispached straight away... you can't abort it like this.

Reply:
maybe we should dispatch in next tick?

Reply:
I can change this

Reply:
Gotcha! Test updated :)

Reply:
> maybe we should dispatch in next tick?

I'm not sure it is worth, also, in Fastify we had some issue in the past by overusing next tick in hot paths. (cc @mcollina)

Reply:
We can make it so we only use it if you send a signal. But I don't think this is such a hot path.

Reply:
I've added the nextTick btw, without a fast path for `!signal`.